neurodamus/utils/memory.py

```diff
@@ -160,6 +161,159 @@ def pretty_printing_memory_mb(memory_mb):
+        # Update total memory and re-add to the heap
+        heapq.heappush(ranks, (total_memory, rank_id))
+
+    # Start distributing cells across ranks
+    for pop, metype_gids in dry_run_stats.metype_gids.items():
+        logging.info("Distributing cells of population %s", pop)
+        rank_allocation[pop] = {}
+        rank_memory[pop] = {}
+        batch = []
+        batch_memory = 0
+
+        for cell_id, memory in generate_cells(metype_gids):
+            batch.append(cell_id)
+            batch_memory += memory
+            if len(batch) == batch_size:
+                # Get the rank with the lowest memory load
+                total_memory, rank_id = heapq.heappop(ranks)
+                logging.debug("Assigning batch to rank %d", rank_id)
+                # Add the cell to the rank
+                assign_cells_to_rank(rank_id, batch_memory, rank_allocation, total_memory,
+                                     rank_memory, pop)
+                batch = []
+                batch_memory = 0
+
+        # Assign any remaining cells in the last, potentially incomplete batch
+        if batch:
+            total_memory, rank_id = heapq.heappop(ranks)
+            logging.debug("Assigning remaining cells in batch to rank %d", rank_id)
+            assign_cells_to_rank(rank_id, batch_memory, rank_allocation, total_memory,
+                                 rank_memory, pop)
+
+    return rank_allocation, rank_memory
+
+
+@run_only_rank0
+def print_allocation_stats(rank_allocation, rank_memory):
+    """
+    Print statistics of the memory allocation across ranks.
+
+    Args:
+        rank_allocation (dict): A dictionary where keys are rank IDs and
+                                values are lists of cell IDs assigned to each rank.
+        rank_memory (dict): A dictionary where keys are rank IDs
+                            and values are the total memory load on each rank.
+    """
+    logging.debug("Rank allocation: {}".format(rank_allocation))
+    logging.debug("Total memory per rank: {}".format(rank_memory))
+    import statistics
+    for pop, rank_dict in rank_memory.items():
+        values = list(rank_dict.values())
+        logging.info("Population: {}".format(pop))
+        logging.info("Mean allocation per rank [KB]: {}".format(round(statistics.mean(values))))
+        try:
+            stdev = round(statistics.stdev(values))
+        except statistics.StatisticsError:
+            stdev = 0
+        logging.info("Stdev of allocation per rank [KB]: {}".format(stdev))
+
+
+@run_only_rank0
+def export_allocation_stats(rank_allocation, filename):
+    """
+    Export allocation dictionary to serialized pickle file.
+    """
+    import pickle
+    try:
+        with open(filename, 'wb') as f:
+            pickle.dump(rank_allocation, f)
+    except Exception as e:
+        logging.warning("Unable to export allocation stats: {}".format(e))
+
+
+@run_only_rank0
+def import_allocation_stats(filename):
```

Comment:
Where is this used?

Reply:
So, at the moment the function is indeed not used anywhere, wrote it as a mirror of the export function for future use when we'll integrate the results into the load balancer. I would delete for the moment, but if you agree, I will at least use it in the workflow test for dryrun.